spec/integration/streams/process_sub_pages_spec.rb

```diff
@@ -0,0 +1,126 @@
+require 'govuk_message_queue_consumer/test_helpers/mock_message'
+
+RSpec.describe "Process sub-pages for multipart content types" do
+  include QualityMetricsHelpers
+
+  let(:subject) { PublishingAPI::Consumer.new }
+
+  before { stub_quality_metrics }
+
+  it "grows the dimension for each subpage" do
+    message = build(:message, :with_parts)
+
+    expect {
+      subject.process(message)
+    }.to change(Dimensions::Item, :count).by(4)
+  end
+
+  it "separates the parts of multipart content types" do
+    message = build(:message, :with_parts)
+    subject.process(message)
+
+    parts = Dimensions::Item.pluck(:base_path, :title).to_set
+
+    expect(parts).to eq Set[
+      ["/base-path/part1", "Part 1"],
+      ["/base-path/part2", "Part 2"],
+      ["/base-path/part3", "Part 3"],
+      ["/base-path/part4", "Part 4"]
+    ]
+  end
+
+  it "can incrementally reprocess messages containing multiple parts, if some of them fail" do
```

Comment:
Not sure if we need to do this, and if the pros/cons to handle this situations is worth from a complexity perspective.